mantidproject/mantid
# Add algorithm GSASIIRefineFitPeaks

Implements an algorithm that uses GSAS-II to refine lattice parameters and fit peaks. Supports Pawley and Rietveld refinement. Uses GSAS-II modules for powder diffraction and structure. 
Not ready for testing. It's missing proper test data.

**To test:**
- Check the documentation of the algorithm
- There's a too simple unit test that just checks that the algorithm is there. No hope of having proper unit/system tests as this requires GSAS-II.
- You need to install a patched version of GSAS-II (ask me for details).
- Requires a GSAS-II instrument parameter file (.par/.prm/.iparm) and a phase information file (.cif), in addition to histogram data.
- Usage example (you can run this script - it will require changing the paths to your local ones, or run the algorithm though its interface where you can browse... for the files):

```
import mantid
from mantid.simpleapi import *
wks_focused = Load('focused_bank1_ENGINX00256663.nxs')
inst_file = 'ENGINX_255924_254854_North_bank.par'
phase_file = 'FE_ALPHA.cif'
gsas_path = '/path/to/gsas/' # replace as needed
GoF, Rwp, lattice_tbl = GSASIIRefineFitPeaks(InputWorkspace=wks_focused,
                                             InstrumentFile=inst_file,
                                             PhaseInfoFile=phase_file,
                                             PathToGSASII=gsas_path,
                                             SaveGSASIIProjectFile='example_gsas2_project',
                                             LatticeParameters='lattice_tbl')

print "Goodness of fit coefficient: {0:.5f}".format(GoF)
print "Weighted profile R-factor (Rwp): {0:.5f}".format(Rwp)
print ("Lattice parameters, a: {a}, b: {b}, c: {c}, alpha: {alpha}, beta: {beta}, gamma: {gamma}, "
       "Volume: {volume:.3f}".format(**lattice_tbl.row(0)))

```
- See also the usage examples in the algorithm documentation page.
- The three required files (phase information, instrument, and focused data) can be found here (ISIS network): `\\olympic\Babylon5\Public\fedemp\ENGINX-X\GSASII_Mantid`
- For the original data files (at ISIS): `\\olympic\babylon5\public\fedemp\ENGINX-X\GSASII_docstests_data`
- If you have installed GSASII, update it it to the latest version from the server: Main menu / Help / Check for updates. This is just to make sure that we're up to date. Then copy the files found in `\\olympic\babylon5\public\fedemp\ENGINX-X\GSASII_Mantid\modified_files` directly to the top level of your GSASII installation (Something like C:... or D:\GSASII). You should see that all these files overwrite existing files. Alternatively, you could apply the patch found in `\\olympic\babylon5\public\fedemp\ENGINX-X\GSASII_Mantid\`.
- If you haven't installed GSASII, for the patched version of GSASII as a whole: `\\olympic\babylon5\public\fedemp\ENGINX-X\GSASII_Mantid\gsas`

Fixes #15771.
Fixes #10630 (an older ticket superseded by this).

Release notes:  https://github.com/mantidproject/mantid/commit/75eb23d98d306ac382e3cfb61c2d270d6d68a859

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
